include/net/net_context.h

```diff
@@ -87,8 +89,10 @@ typedef void (*net_context_recv_cb_t)(struct net_context *context,
  * @typedef net_context_send_cb_t
  * @brief Network data send callback.
  *
- * @details The send callback is called after a network data is
- * sent.
+ * @details The send callback is called after a network data is sent.
+ * This callback is called by TX thread so its stack and execution context is
+ * used here. Do processing inside this callback as fast as possible in order
```

Comment:
Keep processing in the callback minimal to reduce the time spent blocked while handling packets.